pkg/store/kotsstore/supportbundle_store.go

```diff
@@ -275,54 +228,58 @@ func (s *KOTSStore) GetSupportBundle(id string) (*types.SupportBundle, error) {
 		return nil, errors.Wrap(err, "failed to unmarshal")
 	}
 
-	treeindex, err := s.getSupportBundleMetafile(id, "treeindex")
-	if err != nil {
-		if s.IsNotFound(err) {
-			return &supportBundle, nil
+	if supportBundle.Status == "uploaded" {
+		treeindex, err := s.getSupportBundleMetafile(id, "treeindex")
```

Comment:
If this is the only use for "uploaded" status, then we should get rid of it for simplicity.  The call to `s.IsNotFound` below does the same job.

Reply:
No, "Status" is meant to be the key by which the UI decides if a bundle is running, complete or failed. But agree this can come out for simplicity.

Reply:
I see... that's not in this PR